onboarding/source/amb_light_service.c

```diff
@@ -34,20 +46,71 @@ static void lightServiceTask(void * pvParameters);
 obc_error_code_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here. Return error code if task/queue was not created successfully.
+    lightServiceTaskHandle = xTaskCreateStatic( lightServiceTask,
+                                                LIGHT_SERVICE_NAME,
+                                                LIGHT_SERVICE_STACK_SIZE,
+                                                NULL,
+                                                LIGHT_SERVICE_PRIORITY,
+                                                lightServiceTaskStack,
+                                                &lightServiceTaskBuffer);
+    
+    if (lightServiceTaskHandle == NULL) 
+        return OBC_ERR_CODE_TASK_CREATION_FAILED;
+
+    eventQueueHandle = xQueueCreateStatic(  QUEUE_LENGTH,
+                                            QUEUE_ITEM_SIZE,
+                                            eventQueueStorageBuffer,
+                                            &eventQueueBuffer);
+    
+    if (eventQueueHandle == NULL)
+        return OBC_ERR_CODE_QUEUE_CREATION_FAILED;
 
+    return OBC_ERR_CODE_SUCCESS;
     /* USER CODE END */
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    ASSERT(lightServiceTaskHandle != NULL);
+
+    light_event_t lightEventBuffer;
+
+    BaseType_t xCheckEventReturned = xQueueReceive( eventQueueHandle, 
+                                                    &lightEventBuffer, 
+                                                    0);
+
+    if (xCheckEventReturned == pdTRUE) {
+        // Measure the light and print the result
+        adcData_t adcData;
+        adcData_t *adcDataPtr = &adcData;
```

Comment:
No need to declare a separate variable for the pointer. You can just pass `&adcData` into `adcGetData`